### PR TITLE
Adding UC test

### PR DESCRIPTION
**Pull request by sourabhdalvi**
Testing the following
- Ramping Constraints
- Duration Constraints (not all 3 types but only the mixed type)
- Cost formulation with Linear PWL implementation
- Cost formulation with SOS2 implementation (need to add CLP to support SOS2 constraints)

Also needs the PowerSystemTestData branch to be merged for tests to pass

### CODE REVIEW

**Comment by jd-lara** [comment]
looks like for the tests to pass you need to merge the UC Data branch in https://github.com/NREL/PowerSystemsTestData/pull/9

**Comment by sourabhdalvi** [comment]
> looks like for the tests to pass you need to merge the UC Data branch in [NREL/PowerSystemsTestData#9](https://github.com/NREL/PowerSystemsTestData/pull/9)

**Comment by jd-lara** [comment]
@sourabhdalvi looks like since GLPK doesn't support SOS constraints and there aren't bridges the tests can't pass.

**Comment by jd-lara** [comment]
@sourabhdalvi what's the status of this PR?

File: src/devices/device_models/common/cost_functions.jl [hunk 1]
```diff
@@ -176,13 +179,13 @@ function _pwlgencost_sos(canonical_model::CanonicalModel,
                             base_name = "{$(variable)}_{sos}",
                             start = 0.0, lower_bound = 0.0, upper_bound = 1.0)
 
-    sos2 = JuMP.@constraint(canonical_model.JuMPmodel, pwlvars in MOI.SOS2(collect(1:length(pwlvars))))
-
+    JuMP.@constraint(canonical_model.JuMPmodel, sum(pwlvars)  == 1.0)
+    JuMP.@constraint(canonical_model.JuMPmodel, pwlvars in MOI.SOS2(collect(1:length(pwlvars))))
```
Comment (jd-lara) [inline src/devices/device_models/common/cost_functions.jl]: is this correct `length(pwlvars)` = `l`?

**Pull request merged**